pkg/resource/broker.go

```diff
@@ -0,0 +1,82 @@
+package resource
+
+import (
+	"context"
+
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"go.uber.org/zap"
+)
+
+type ID string
+
+type FileWriter interface {
+	storage.ExternalFileWriter
```

Comment:
Does DM have the `flush file` requirement, `storage.ExternalFileWriter` doesn't have one.

Reply:
Seems Close is enough

// Close writes final chunk and completes the upload.